src/main.rs

```diff
@@ -234,26 +241,39 @@ fn update_ncube_meshes(
 
 fn rotate_ncube(
     time: Res<Time>,
+    ncube_planes_of_rotation: Res<NCubePlanesOfRotation>,
+    ncube_is_paused: Res<NCubeIsPaused>,
     mut ncube: ResMut<NCube>,
     mut ncube_rotations: ResMut<NCubeRotations>,
-    ncube_planes_of_rotation: Res<NCubePlanesOfRotation>,
     mut ncube_vertices_3d: ResMut<NCubeVertices3D>,
-    ncube_is_paused: Res<NCubeIsPaused>,
+    mut ncube_correction: ResMut<NCubeCorrection>,
 ) {
     if **ncube_is_paused {
         return;
     }
-    let dt = time.delta_seconds();
+    if ncube_correction.current_angle > std::f64::consts::TAU {
+        println!("APPLIED CORRECTION");
+        **ncube_rotations = ncube_correction.rotations.clone();
+        **ncube_vertices_3d = ncube_correction.vertices_3d.clone();
+        ncube.vertices = ncube_correction.vertices.clone();
+        ncube_correction.reset_angle();
```

Comment:
This sets the angle to `0`, but I think it should subtract `TAU`. This might be one reason for the jump.

Reply:
That doesn't affect it since the jump even happens during the first full rotation.
If you test it yourself (by pulling this branch and running `make run`) you'll notice that the only cases with a jump are those where rotation planes overlap.